ext/net/win_pipe.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+use std::borrow::Cow;
+use std::ffi::OsStr;
+use std::io;
+use std::rc::Rc;
+
+use deno_core::AsyncRefCell;
+use deno_core::AsyncResult;
+use deno_core::CancelHandle;
+use deno_core::CancelTryFuture;
+use deno_core::RcRef;
+use deno_core::Resource;
+use tokio::io::AsyncReadExt;
+use tokio::io::AsyncWriteExt;
+use tokio::net::windows::named_pipe;
+
+pub struct NamedPipe {
+  inner: AsyncRefCell<Inner>,
+  cancel: CancelHandle,
+}
+
+enum Inner {
+  Server((bool, named_pipe::NamedPipeServer)),
+  Client(named_pipe::NamedPipeClient),
+}
+
+impl NamedPipe {
+  pub fn new_server(
+    addr: impl AsRef<OsStr>,
+    options: &named_pipe::ServerOptions,
+  ) -> io::Result<NamedPipe> {
+    let server = options.create(addr)?;
+    // todo: should we connect here
+    Ok(NamedPipe {
+      inner: AsyncRefCell::new(Inner::Server((false, server))),
+      cancel: Default::default(),
+    })
+  }
+
+  pub fn new_client(
+    addr: impl AsRef<OsStr>,
+    options: &named_pipe::ClientOptions,
+  ) -> io::Result<NamedPipe> {
+    let client = options.open(addr)?;
+    Ok(NamedPipe {
+      inner: AsyncRefCell::new(Inner::Client(client)),
+      cancel: Default::default(),
+    })
+  }
+
+  pub async fn write(self: Rc<Self>, buf: &[u8]) -> io::Result<usize> {
+    let mut inner = RcRef::map(&self, |s| &s.inner).borrow_mut().await;
+    let cancel = RcRef::map(&self, |s| &s.cancel);
+    match &mut *inner {
+      Inner::Server((ref mut connected, server)) => {
+        if !*connected {
+          server.connect().await?;
```

Comment:
Why wait to connect until read/write is called? I assume it's related to the todo in `new_server`?